src/plugins/blackbox/public/js/blackbox.js

```diff
@@ -637,23 +637,27 @@
     .then(function (session) {
       session_record = session;
       var sort = 0;
+      var promiseChain = Promise.resolve();
       self.idb.telemetry_events
         .where('sessionID').equalsIgnoreCase(options.sessionID)
         .filter(function (item) {
           return item.event == 'x-h264-video.data';
         })
         .each(function(videoItem){
-          return new Promise(function(resolve,reject){
-            save(new Uint8Array(videoItem.data),function(){
-              resolve();
-            }); 
+          promiseChain = promiseChain.then(function(){
```

Comment:
This was a fix for using the each iterator to build a new promise chain.  The each function does not actually collect the results from the execution.